File: forge/src/main/java/io/github/xiewuzhiying/vs_addition/forge/compat/computercraft/PeripheralForge.java
//package io.github.xiewuzhiying.vs_addition.forge.compat.computercraft;
//
//import dan200.computercraft.api.peripheral.IPeripheral;
//import io.github.xiewuzhiying.vs_addition.forge.compat.computercraft.peripherals.CompactCannonMountPeripheral;
//import net.minecraft.core.BlockPos;
//import net.minecraft.core.Direction;
//import net.minecraft.world.level.Level;
//import net.minecraft.world.level.block.Block;
//import net.minecraft.world.level.block.entity.BlockEntity;
//import net.minecraft.world.level.block.state.BlockState;
//import riftyboi.cbcmodernwarfare.cannon_control.compact_mount.CompactCannonMountBlockEntity;
//import riftyboi.cbcmodernwarfare.index.CBCModernWarfareBlocks;
//
//import javax.annotation.Nullable;
//
//public class PeripheralForge {
//
//    private boolean c(BlockState arg1, Block arg2) { return arg1.getBlock() == arg2; }
//
//    @Nullable
//    public IPeripheral getPeripheralForge(Level level, BlockPos blockPos, Direction direction){
//        BlockState s = level.getBlockState(blockPos);
//        BlockEntity be = level.getBlockEntity(blockPos);
//        if (c(s, CBCModernWarfareBlocks.COMPACT_MOUNT.get())) {
//            return new CompactCannonMountPeripheral("cbcmf_compact_cannon_mount", (CompactCannonMountBlockEntity) be, level, blockPos, direction);
//        } else {
//            return null;
//        }
//    }
//}

File: forge/src/main/kotlin/io/github/xiewuzhiying/vs_addition/forge/compats/computercraft/PeripheralForge.kt
package io.github.xiewuzhiying.vs_addition.forge.compats.computercraft

import dan200.computercraft.api.ComputerCraftAPI
import dan200.computercraft.api.peripheral.IPeripheral
import dev.architectury.platform.Platform
import io.github.xiewuzhiying.vs_addition.compats.computercraft.PeripheralCommon.PeripheralSupplier
import io.github.xiewuzhiying.vs_addition.forge.compats.computercraft.peripherals.CheatCompactCannonMountMethods
import io.github.xiewuzhiying.vs_addition.forge.compats.computercraft.peripherals.CompactCannonMountMethods
import net.minecraft.core.BlockPos
import net.minecraft.core.Direction
import net.minecraft.world.level.Level
import net.minecraft.world.level.block.Block

object PeripheralForge {
    private val peripheralMap: MutableMap<Block, PeripheralSupplier> = HashMap()

    fun getPeripheralForge(level: Level, blockPos: BlockPos, direction: Direction?): IPeripheral? {
        val s = level.getBlockState(blockPos)
        val be = level.getBlockEntity(blockPos)
        val supplier = peripheralMap[s.block]
        return be?.let { supplier?.get(it, level, blockPos) }
    }

    @JvmStatic
    fun registerGenericPeripheralForge() {
        if (Platform.isModLoaded("cbcmodernwarfare")) {
            ComputerCraftAPI.registerGenericSource(CompactCannonMountMethods());
            ComputerCraftAPI.registerGenericSource(CheatCompactCannonMountMethods());
        }
    }
}